services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_events_user_services.py

```diff
@@ -73,6 +73,9 @@ async def submit_compose_sepc(app: FastAPI, scheduler_data: SchedulerData) -> No
     assert (
         scheduler_data.product_name is not None  # nosec
     ), "ONLY for legacy. This function should not be called with product_name==None"
+    assert (
+        scheduler_data.product_api_base_url is not None  # nosec
+    ), "ONLY for legacy. This function should not be called with product_api_base_url==None"
```

Comment:
is this use-case really needed? should it not just be non nullable from the start? what is the use-case when it is None?
same with the one for the product_name. does it even happen?